File: app/src/main/java/com/example/eziketobenna/bakingapp/ui/recipes/RecipeViewModel.java
package com.example.eziketobenna.bakingapp.ui.recipes;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.example.eziketobenna.bakingapp.data.RecipeRepository;
import com.example.eziketobenna.bakingapp.data.model.Recipe;

import java.util.List;

import javax.inject.Inject;

/**
 * {@link ViewModel} for {@link RecipeFragment}
 */
class RecipeViewModel extends ViewModel {

    private RecipeRepository mRecipeRepository;

    private final LiveData<List<Recipe>> mAllRecipes;

    @Inject
    RecipeViewModel(RecipeRepository repository) {
        mRecipeRepository = repository;
        mAllRecipes = repository.getAllRecipes();
    }

    LiveData<List<Recipe>> getAllRecipes() {
        return mAllRecipes;
    }

    void retryFetch() {
        mRecipeRepository.retryFetch();
    }
}

File: app/src/main/java/com/example/eziketobenna/bakingapp/ui/recipes/RecipeViewModel.kt
package com.example.eziketobenna.bakingapp.ui.recipes

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.eziketobenna.bakingapp.data.RecipeRepository
import com.example.eziketobenna.bakingapp.data.model.Recipe
import javax.inject.Inject

class RecipeViewModel @Inject constructor(
    private val mRecipeRepository: RecipeRepository
) : ViewModel() {

    val _allRecipes: MutableLiveData<List<Recipe>> = MutableLiveData()
    val allRecipes: LiveData<List<Recipe>> by this::_allRecipes

    fun retryFetch() {
        mRecipeRepository.retryFetch()
    }
}
